core/templates/pages/admin-page/services/admin-misc-tab-backend-api.service.spec.ts

```diff
@@ -0,0 +1,198 @@
+        console.log(failHandler)
+      }
+      ));
+      
+      it('should upload topic similarities',
+      fakeAsync(() => {
+        let data = 'topic_similarities.csv';
+        adminMiscTabBackendApiService.uploadTopicSimilarities(data)
+          .then(successHandler, failHandler);
+  
+        let req = httpTestingController.expectOne(
+          '/adminhandler');
+        expect(req.request.method).toEqual('POST');
+        req.flush(200);
+        flushMicrotasks();
+  
+        expect(successHandler).toHaveBeenCalled();
+        expect(failHandler).not.toHaveBeenCalled();
+      }
+      ));
+      
+      it('should send dummy mail to admin',
+      fakeAsync(() => {
+        let errorMessage='Server error: This app cannot send emails.'
+        adminMiscTabBackendApiService.sendDummyMail()
+          .then(successHandler, failHandler);
+  
+        let req = httpTestingController.expectOne(
+          '/senddummymailtoadminhandler');
+        expect(req.request.method).toEqual('POST');
+        req.flush(
+          { error: errorMessage },
+          { status: 400, statusText: ''}
+        );
+        flushMicrotasks();
+  
+        expect(successHandler).not.toHaveBeenCalled();
+        expect(failHandler).toHaveBeenCalled();
+      }
+      ));
+
+      it('should get data of memory cache profile',
+      fakeAsync(() => {
+        adminMiscTabBackendApiService.getMemoryCacheProfile()
+          .then(successHandler, failHandler);
+  
+        let req = httpTestingController.expectOne(
+          '/memorycacheadminhandler');
+        expect(req.request.method).toEqual('GET');
+        req.flush(200);
+        flushMicrotasks();
+  
+        expect(successHandler).toHaveBeenCalled();
+        expect(failHandler).not.toHaveBeenCalled();
+      }
+      ));
+
+      it('should update the username of oppia account',
+      fakeAsync(() => {
+        let oldUsername = 'old name'
+        let newUsername = 'new name'
+        adminMiscTabBackendApiService.updateUserName(oldUsername,newUsername)
+          .then(successHandler, failHandler);
+  
+        let req = httpTestingController.expectOne(
+          '/updateusernamehandler');
+        expect(req.request.method).toEqual('PUT');
+        req.flush(200);
+        flushMicrotasks();
+  
+        expect(successHandler).toHaveBeenCalled();
+        expect(failHandler).not.toHaveBeenCalled();
+      }
+      ));
+      
+      it('should get the data of number of pending delete requests',
+      fakeAsync(() => {
+        adminMiscTabBackendApiService.numberOfPendingDeletionRequest()
+          .then(successHandler, failHandler);
+  
+        let req = httpTestingController.expectOne(
+          '/numberofdeletionrequestshandler');
+        expect(req.request.method).toEqual('GET');
+        req.flush(200);
+        flushMicrotasks();
+  
+        expect(successHandler).toHaveBeenCalled();
+        expect(failHandler).not.toHaveBeenCalled();
+      }
+      ));
+  });
```

Comment:
Add newline (can you please confirm that the pre-push hooks are working for you?)

Reply:
> Add newline (can you please confirm that the pre-push hooks are working for you?)

They have not triggered automatically. Any suggestions on how to resolve this?

Reply:
They are working fine now.